aws/elastic-beanstalk/elastic-beanstalk.tf

```diff
@@ -0,0 +1,594 @@
+  default     = 1
+  description = "Maximum number of instances you want in your Auto Scaling group"
+}
+
+variable "asg_availability_zones" {
+  type        = string
+  default     = "any"
+  description = "Availability Zones (AZs) are distinct locations within a region that are engineered to be isolated from failures in other AZs and provide inexpensive, low-latency network connectivity to other AZs in the same region. Choose the number of AZs for your instances"
+}
+
+variable "environment_environment_type" {
+  type        = string
+  default     = "LoadBalanced"
+  description = "Set to SingleInstance to launch one EC2 instance with no load balancer."
+}
+
+variable "environment_load_balancer_type" {
+  type        = string
+  default     = "application"
+  description = "The type of load balancer for your environment"
+}
+
+variable "alb_cert" {
+  description = "Certificate attached to ALB"
+  default     = null
+  type        = string
+}
+
+variable "default_process_port" {
+  type        = number
+  default     = 80
+  description = "Port on which the process listens"
+}
+
+variable "launchconfiguration_ec2_key_name" {
+  type        = string
+  default     = null
+  description = "A key pair enables you to securely log into your EC2 instance"
+}
+
+variable "default_process_protocol" {
+  type        = string
+  default     = "HTTP"
+  description = "Protocol that the process uses. With an application load balancer, you can only set this option to HTTP or HTTPS. With a network load balancer, you can only set this option to TCP."
+}
+
+variable "vpc_id" {
+  default = null
+  type    = string
+}
+
+variable "alb_subnets" {
+  default = null
+  type    = list
+}
+
+variable "ec2_subnets" {
+  default = null
+  type    = list
+}
+
+variable "instances_instance_types" {
+  type        = string
+  default     = "t3.medium"
+  description = "A comma-separated list of instance types you want your environment to use. For example: t2.micro,t3.micro"
+}
+
+variable "aliases" {
+  type        = list(string)
+  description = "List of FQDN's - Used to set the Alternate Domain Names (CNAMEs) setting on Cloudfront"
+  default     = []
+}
+
+variable "parent_zone_id" {
+  type        = string
+  default     = null
+  description = "ID of the hosted zone to contain this record  (or specify `parent_zone_name`)"
+}
+
+variable "parent_zone_name" {
+  type        = string
+  default     = null
+  description = "Name of the hosted zone to contain this record (or specify `parent_zone_id`)"
+}
+
+variable "https_redirect" {
+  type        = bool
+  default     = false
+  description = "When ALB is used, this will create a listener to redirect http to https"
+}
+
+# ----------------------------------------------------------------------------------------------------------------------
+# MODULES / RESOURCES
+# ----------------------------------------------------------------------------------------------------------------------
+
+resource "aws_iam_role" "ec2" {
```

Comment:
When naming resources, put the name of the file in the resources (e.g. `eb_ec2` from `eb.tf`). This is helpful in case you leverage the same resource across different areas of the module.